pkg/engine/jmespath/functions.go

```diff
@@ -550,6 +579,53 @@ func jpBase64Encode(arguments []interface{}) (interface{}, error) {
 	return base64.StdEncoding.EncodeToString([]byte(str.String())), nil
 }
 
+func jpTimeSince(arguments []interface{}) (interface{}, error) {
+	var err error
+	layout_, err := validateArg("", arguments, 0, reflect.String)
+	if err != nil {
+		return nil, err
+	}
+
+	layout := layout_.String()
```

Comment:
Variable naming in this entire section is far from ideal - why are we putting `layout.String()` into a separate variable?
Also variable naming with underscores is not usually done in golang.

Reply:
`layout` and `layout.String()` have different types so we need a separate variable to handle this. I'll change `layout_` to `temp`.

Reply:
You can just use `layout.String()` whenever you need to invoke the string value. 

Same with the variables below:
```
timestamp1, err := validateArg("", arguments, 1, reflect.String)
if err != nil {
return nil, err
}
t1, err := time.Parse(layout, timestamp1.String())
if err != nil {
return nil, err
}
```

Reply:
Got it. If I remove the `layout_` variable, I'd need to add two more key checks for the `layouts` map for parsing timestamps for `t1` and `t2`. Saving the string in a temporary variable and updating if it's present in the map once, saves us from one potential extra check which would be costlier than copying the string.